tests/impute/test_impute_mode.py

```diff
@@ -0,0 +1,25 @@
+from chispa import assert_df_equality
+
+from cishouseholds.impute import calculate_imputation_from_mode
+
+
+def test_impute_distribution(spark_session):
+
+    expected_df = spark_session.createDataFrame(
+        data=[
+            # current stata uses 1, 2 for "male", "female"
+            ("E12000007", "white", "male", None),  # proportion white-male = 1.0 therefore all rnd < 1.0
+            ("E12000007", "white", "male", None),
+            ("E12000007", None, None, "white"),
+            ("E12000007", "white", None, None),
```

Comment:
The grouping needs to be done by household, so the test data needs a household indicator, the GOR9D is not needed.

Reply:
Grouping are possible to be selected. I have a purpose oriented data to test all the possible exceptions to make sure it passes correctly.